hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/MockNodeManager.java

```diff
@@ -398,6 +399,22 @@ public RegisteredCommand register(DatanodeDetails datanodeDetails,
     return null;
   }
 
+  /**
+   * Add an entry to the dnsToUuidMap, which maps hostname / IP to the DNs
+   * running on that host. As each address can have many DNs running on it,
+   * this is a one to many mapping.
+   * @param dnsName String representing the hostname or IP of the node
+   * @param uuid String representing the UUID of the registered node.
+   */
+  private void addEntryTodnsToUuidMap(String dnsName, String uuid) {
```

Comment:
It also can by synchronized and HashSet - > ConcurrentHashSet.

Reply:
I have fixed this one by creating a set with ConcurrentHashMap.newKeySet(); which creates a Concurrent set and synchronised the method.